AdventOfCode2023/Day6.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Linq;
+
+namespace AdventOfCode2023
+{
+    internal class Day6
+    {
+        public static int Day6Main(string[] input)
+        {
+            var times = input[0].Split(':')[1].Trim().Split(new char[] {' '}, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse).ToArray();
+            var distances = input[1].Split(':')[1].Trim().Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse).ToArray();
+            int winOptionProduct = 1;
+            
+            for (var i = 0; i < times.Length; i++)
+            {
+                winOptionProduct *= CheckRaceWinOptions(times[i], distances[i]);
+            }
+
+            return winOptionProduct;
+        }
+
+        internal static int CheckRaceWinOptions(int raceTime, int distanceToBeat)
```

Comment:
using linq the whole thing between line 22 and 35 can be converted to 

internal static int CheckRaceWinOptions(int raceTime, int distanceToBeat)
{
    return Enumerable.Range(0, raceTime + 1)
                     .Count(holdSec => (raceTime - holdSec) * holdSec > distanceToBeat);
}